### DISCUSSION TITLE
Reverse C# scripting

### DISCUSSION THREAD

**Discussion by rvgfox** [TabularEditor/TabularEditor3#836] [🙋♀️TE3 Q&A]
With C# scripting we can create columns, measures and so on in the model.

My question it's, if is it possible to create a C# script from an existing column or measure in order to save it and replicate in other models?

**Comment by mlonsk** [comment]
Hi @rvgfox 
You can record scripts through the C# Script editor, as shown bellow. 
Is that what you were looking for?

![RecordingScripts](https://user-images.githubusercontent.com/11939114/222374933-ca07e56b-3c32-42ef-b401-1d6b49593b29.gif)

**Reply by mlonsk** [reply]
I have moved this to our discussion section, as it fits better here. Hope that is okay.

**Comment by rvgfox** [comment]
No, the idea it's to with an existing measure with it's DAX expression ant the rest of properties, get the C# script to create it.

**Comment by mlonsk** [comment]
Ah okay, there I think the Script DAX functionality is what you are looking for, if I understand you correctly. 
Right click on the measure and choose Script DAX, then save the DAX Script and reopen in another model and execute the DAX Script there. 
![ScriptingDAX](https://user-images.githubusercontent.com/11939114/222406123-8030f637-9bbc-41f4-8920-b38cb7efc0f9.gif)


It is, currently, not possible to execute DAX Scripts in a C# script, but it is something we have on our backlog to provede APIs for.

**Comment by rvgfox** [comment]
The idea it's "recreate" the C# script to create the measure and use it in another model
